### PR TITLE
[11.0[IMP] l10n_nl_xaf_auditfile_export: Memory footprint, speed and disk space

### PR DESCRIPTION
**Pull request by astirpe**
I'm experimenting the forward-port of #167 (by @StefanRijnhart ) to V11 with some adaptations.

Depends on:

- [x] PR #182 (for field `auditfile_name` stored + tests)

### CODE REVIEW

File: l10n_nl_xaf_auditfile_export/tests/test_l10n_nl_xaf_auditfile_export.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import base64
+from io import BytesIO
+from zipfile import ZipFile
+
+from odoo.tools import mute_logger
+from odoo.tests.common import TransactionCase
+
+
+class TestXafAuditfileExport(TransactionCase):
```
Comment (StefanRijnhart) [inline l10n_nl_xaf_auditfile_export/tests/test_l10n_nl_xaf_auditfile_export.py]: Did the port of this module not include the tests from 8.0 that I am extending in https://github.com/OCA/l10n-netherlands/pull/167/files? Do you not want to take these tests that validate the zip format of the resulting file and the xml format of the zip file's contest?

Comment (astirpe) [reply l10n_nl_xaf_auditfile_export/tests/test_l10n_nl_xaf_auditfile_export.py]: That's exactly what is included in commit https://github.com/OCA/l10n-netherlands/pull/183/commits/c6d3b6ed13e7f0b1b509445d81bff2a023be1933. Is this what you mean?

Comment (astirpe) [reply l10n_nl_xaf_auditfile_export/tests/test_l10n_nl_xaf_auditfile_export.py]: https://github.com/OCA/l10n-netherlands/commit/c6d3b6ed13e7f0b1b509445d81bff2a023be1933#diff-5a1c740bcb76f9e92b4e053873f60f21

Comment (StefanRijnhart) [reply l10n_nl_xaf_auditfile_export/tests/test_l10n_nl_xaf_auditfile_export.py]: Yes, sorry I missed that!

**Review by StefanRijnhart — APPROVED** [review]

**Comment by astirpe** [comment]
Branch rebased.
We are using this version of the module internally and it seems working fine

**Comment by astirpe** [comment]
@yung-wang

**Review by yung-wang — APPROVED** [review]
tested and file is much smaller and the export is also faster

**Pull request merged**